<h1> Applications in alphabetical order</h1>
<!-- head -2 alpha.md > tmp; grep \* index.md | sort | uniq >> tmp;mv tmp alpha.md -->
* [Abaqus](abaqus.md) Dassault Systemes' SIMULIA academic research suite
* [Amber](amber.md) Molecular dynamics suite
* [AMS](ams.md) Modelling suite providing engines like ADF, BAND, DFTB and MOPAC
* [ANSYS](ansys.md) ANSYS Academic engineering simulation software suite
* [ArcGIS](arcgis.md) ArcGIS software suite for working with spatial data 
* [BAMtools](bamtools.md) Tools for working with BAM formatted files
* [Bioconda](bioconda.md) Package manger for installing bioinformatics software
* [BioPerl](bioperl.md) Perl environment with bioperl extension
* [BioPython](biopython.md) Pyhton environment with biopython and other bioinformatics related Python libraries
* [BLAST](blast.md) Sequence similarity search tool for nucleotides and proteins
* [Bowtie2](bowtie2.md) Short read aligner 
* [BWA](bwa.md) Short read aligner
* [CD-hit](cd-hit.md) Sequence clustering and redundancy removal tool
* [CFDEMcoupling](cfdem.md) Open source toolbox for CFD-DEM coupling
* [COMSOL Multiphysics](comsol.md) General-purpose simulation software
* [CP2k](cp2k.md) DFT, quantum chemistry, QM/MM, AIMD etc. in particular for periodic systems
* [DDT](ddt.md) Parallel debugger
* [Diamond](diamond.md ) Sequence similarity search tool for proteins and nucloeotides
* [Edirect](edirect.md) Entrez direct - command line tool to search and retrieve data from NCBI
* [Elmer](elmer.md) Open source multi-physics FEM package
* [EMBOSS](emboss.md) Toolkit for classical sequence analysis
* [Exonerate](exonerate.md) A generic tool for pairwise sequence comparison.
* [FastQC](fastqc.md) Quality control tool for high throughput sequence data
* [finnish-parse](http://urn.fi/urn:nbn:fi:lb-2017030801) Dependency Parser for Finnish
* [Finnish Tagtools](http://urn.fi/urn:nbn:fi:lb-201908161) Finnish Tagtools 1.4
* [FORCE](force.md) All-in-one command line tool for mass-processing of medium-resolution satellite images
* [Freebayes](freebayes.md) Genetic variant detector
* [Gaussian](gaussian.md) Versatile computational chemistry package
* [GDAL](gdal.md) Translator library for geospatial data formats
* [Geoconda](geoconda.md) Spatial analysis libraries for Python, QGIS, GDAL and LasTools
* [GPAW](gpaw.md) Versatile DFT package
* [Gromacs](gromacs.md) Fast and versatile classical molecular dynamics
* [HFST-fi](http://urn.fi/urn:nbn:fi:lb-201509034) Helsinki Finite-State Technology for Finnish
* [HFST](http://urn.fi/urn:nbn:fi:lb-20140730183) Helsinki Finite-State Transducer Technology
* [HFST-sv](http://urn.fi/urn:nbn:fi:lb-201509035) Helsinki Finite-State Technology for Swedish
* [IDL](idl.md) Programming Language, Numeric Analysis, Manipulation and Visualization of Scientific Data
* [Illumina BaseSpace](bs.md) Command line client for retrieving data from the Illumina BaseSpace environment
* [Ipyrad](ipyrad.md) toolkit for population genetic and phylogenetic studies of restriction-site associated genomic data sets (e.g., RAD, ddRAD, GBS)
* [Julia](julia.md)  High-level, high-performance dynamic programming language for numerical computing
* [Kaldi](http://urn.fi/urn:nbn:fi:lb-2019121020) Kaldi Speech Recognition Toolkit
* [LAStools](lastools.md) Toolbox for LiDAR datasets
* [LS-DYNA](lsdyna.md) General-purpose finite element program
* [MACS2](macs2.md) ChIP-Seq analysis tool
* [Maestro](maestro.md) Versatile drug discovery and materials modeling suite
* [Mapnik](mapnik.md) Open source toolkit for rendering maps
* [MATLAB](matlab.md) High-level technical computing language
* [Megahit](megahit.md) Metagenomics assebly
* [Minimap2](minimap2.md) Short read aligner
* [MIRA](mira.md) Whole genome shotgun and EST sequence Assembler
* [MOLDEN](molden.md) Processing program for molecular and electronic structure calculations
* [MOLPRO](molpro.md) Package for accurate ab initio quantum chemistry calculations
* [MXNet](mxnet.md) Deep learning library for Python
* [MySQL ](../data/kaivos/kaivos.md) MySQL database client 
* [NAMD](namd.md) Highly scalable classical molecular dynamics
* [NoMachine](nomachine.md) Remote Desktop enabling fast graphics without X-emulators
* [NWChem](nwchem.md) A computational chemistry software package designed to perform well on parallel HPC systems.
* [Octave](octave.md) High-level interpreted language for numerical computations.
* [Open Babel](openbabel.md) Program to interconvert file formats currently used in molecular modeling
* [OpenDroneMap](opendronemap.md) open source command line toolkit for processing aerial drone imagery
* [OpenFOAM](openfoam.md) Open source C++ tool box for continuum mechanics problems
* [ORCA](orca.md) General purpose quantum chemistry package
* [Orfeo ToolBox](otb.md) Open source toolbox for processing high resolution optical, multispectral and radar images
* [PALM](palm.md) Meteorological model system for atmospheric and oceanic boundary-layer flows
* [ParaView](paraview.md) Free open-source visualization application
* [PDAL](pdal.md) open source command line application for point cloud translations and processing
* [Python Data](python-data.md) Collection of Python libraries for data analytics and machine learning
* [PyTorch](pytorch.md) Machine learning framework for Python
* [QGIS](qgis.md) GIS application for viewing, editing, and analysing geospatial data
* [QIIME2](qiime.md) Package for microbial community analysis of amplicon sequencing data
* [RAPIDS](rapids.md) Suite of libraries for data analytics and machine learning on GPUs
* [R for GIS](r-env-for-gis.md) Documentation on the GIS libraries for R
* [R](r-env.md) Open-source language and environment for statistical analysis and graphics
* [RStudio](rstudio.md) Integrated development environment for R
* [SAGA GIS](saga-gis.md) GIS application for spatial data editing and analysis
* [SageMath](sagemath.md) Free open-source mathematics software system
* [Salmon](salmon.md) Program to produce transcript-level quantification estimates from RNA-seq data
* [Samtools](samtools.md) Utilities for managing SAM/BAM formatted alignment files
* [Sen2Cor](sen2cor.md) Stand-alone processor for the Sentinel-2 satellite
* [Seqtk](seqtk.md) Tool for processing sequences in the FASTA or FASTQ format
* [SNAP](snap.md) Remote sensing toolbox for satellites
* [Sofi3d](sofi3D.md) Application for 3D Finite-Difference Seismic Wave Simulation
* [Solaris](solaris.md) Open source deep learning pipeline for geospatial imagery
* [SPAdes](spades.md) Genome assembly
* [Spark](spark.md) High-performance distributed computing framework
* [Stacks](stacks.md) Pipeline for building loci from short-read sequences (e.g. RAD-seq data)
* [STAR](star.md) Short read aligner
* [Structure](structure.md) Inference of population structure in genetics
* [TensorFlow](tensorflow.md) Deep learning library for Python
* [TmoleX](tmolex.md) GUI for setting up and analyzing TURBOMOLE jobs
* [Trinity](trinity.md) Traskriptome assembly tool
* [Turbomole](turbomole.md) Efficient program package for electronic structure calculations
* [UDPipe](http://urn.fi/urn:nbn:fi:lb-201902131) UDPipe Kielipankki version
* [VASP](vasp.md) Ab initio DFT electronic structures
* [Velvet](velvet.md) Genome assembler
* [VisIt](visit.md) Free open-source visualization application
* [VMD](vmd.md) Molecular visualization program 
* [WhiteboxTools](whiteboxtools.md) an advanced geospatial data analysis platform
* [wtdbg2](wtdbg2.md) Fast assembler for long-read data
* [Zonation](zonation.md) Spatial conservation prioritization framework 
