-e # Applications

-e - [By discipline](by_discipline.md)
-e - [By availability](by_system.md)
-e - [By license](by_license.md)
-e 
## Applications in alphabetical order

* [Abaqus](abaqus.md) Dassault Systemes' SIMULIA academic research suite
* [Accelerated visualization](accelerated-visualization.md) A selection of GPU accelerated visualization applications
* [AlphaFold](alphafold.md) Protein 3D structure prediction
* [Amber](amber.md) Molecular dynamics suite
* [Ames Stereo Pipeline](ames-stereo.md) for processing stereo images
* [AMS](ams.md) Modelling suite providing engines like ADF, BAND, DFTB and MOPAC
* [AMS-GUI](ams-gui.md) AMS integrated GUI
* [ANSYS](ansys.md) ANSYS Academic engineering simulation software suite
* [ArcGIS](arcgis.md) General purpose GIS software family for viewing, editing and analysing geospatial data
* [BAMtools](bamtools.md) Tools for working with BAM formatted files
* [BayeScan](bayescan.md) Tool for identifying candidate loci under natural selection based on allele frequencies in populations
* [Bioconda](bioconda.md) Package manger for installing bioinformatics software
* [BioPerl](bioperl.md) Perl environment with bioperl extension
* [BioPython](biopython.md) Python environment with biopython and other bioinformatics related Python libraries
* [BLAST](blast.md) Sequence similarity search tool for nucleotides and proteins
* [Bowtie2](bowtie2.md) Short read aligner
* [BRAKER](braker.md) Automatic genome annotator for eucaryots
* [BWA](bwa.md) Short read aligner
* [CD-hit](cd-hit.md) Sequence clustering and redundancy removal tool
* [Chipster_genomes](chipster_genomes.md) Tool to download aligner indexes used by Chipster to Puhti
* [Chipster](https://chipster.csc.fi/) Easy-to-use analysis platform for RNA-seq, single cell RNA-seq and other NGS data
* [CloudCompare](cloudcompare.md) for visualizing, editing and processing poing clouds
* [COMSOL Multiphysics](comsol.md) General-purpose simulation software
* [COSMO-RS](cosmors.md) Toolbox for predictive property calculation of liquids
* [CP2k](cp2k.md) DFT, quantum chemistry, QM/MM, AIMD etc. in particular for periodic systems
* [CSD](csd.md) Cambridge Crystallographic Database - organic and metallo-organic crystal structures and tools
* [CUDA Quantum](cuda-quantum.md) open-source platform for hybrid quantum-classical computing
* [Cutadapt](cutadapt.md) Trimming high-throughput sequencing reads
* [DDT](ddt.md) Parallel debugger
* [Desktop](desktop.md) Remote desktop environment
* [Diamond](diamond.md ) Sequence similarity search tool for proteins and nucloeotides
* [Discovery Studio](discovery-studio.md) Protein modeling environment
* [eBay's tsv-tools](http://urn.fi/urn:nbn:fi:lb-202006081) Utilities for manipulating large tabular data files
* [Edirect](edirect.md) Entrez direct - command line tool to search and retrieve data from NCBI
* [Elmer](elmer.md) Open source multi-physics FEM package
* [EMBOSS](emboss.md) Toolkit for classical sequence analysis
* [Exonerate](exonerate.md) A generic tool for pairwise sequence comparison
* [FastQC](fastqc.md) Quality control tool for high throughput sequence data
* [finnish-parse](http://urn.fi/urn:nbn:fi:lb-2017030801) Dependency Parser for Finnish
* [Finnish Tagtools](http://urn.fi/urn:nbn:fi:lb-201908161) Finnish Tagtools
* [FORCE](force.md) for mass-processing of medium-resolution satellite images
* [Freebayes](freebayes.md) Genetic variant detector
* [Gaussian](gaussian.md) Versatile computational chemistry package
* [GDAL](gdal.md) for geospatial data formats
* [Geoconda](geoconda.md) - Python spatial analysis libraries
* [GOLD](gold.md) Protein Ligand Docking Software
* [GPAW](gpaw.md) Versatile DFT package
* [Grace](grace.md) Plotting tool for xvg-files in particular
* [GRASS GIS](grass.md) General purpose GIS software family for viewing, editing and analysing geospatial data
* [Gromacs](gromacs.md) Fast and versatile classical molecular dynamics
* [HFST-fi](http://urn.fi/urn:nbn:fi:lb-201509034) Helsinki Finite-State Technology for Finnish
* [HFST](http://urn.fi/urn:nbn:fi:lb-20140730183) Helsinki Finite-State Transducer Technology
* [HFST-sv](http://urn.fi/urn:nbn:fi:lb-201509035) Helsinki Finite-State Technology for Swedish
* [HMMER](hmmer.md) Toolkit to create and use sequence profile hidden Markov models
* [HUMAnN](humann.md) Profiling microbial pathways with metagenomic data
* [HyperQueue](hyperqueue.md) Scheduler for sub-node tasks
* [IDL](idl.md) Programming Language, Numeric Analysis, Manipulation and Visualization of Scientific Data
* [Illumina BaseSpace](bs.md) Command line client for retrieving data from the Illumina BaseSpace environment
* [InterProScan](interpro.md) Protein signature/motif search tool
* [Ipyrad](ipyrad.md) toolkit for population genetic and phylogenetic studies of restriction-site associated genomic data sets (e.g., RAD, ddRAD, GBS)
* [JAX](jax.md) Autograd and XLA, brought together for high-performance machine learning
* [Julia](julia.md)  High-level, high-performance dynamic programming language for numerical computing
* [Julia-Jupyter](julia-jupyter.md) Interactive computational environment with Julia
* [Jupyter for courses](jupyter-for-courses.md) A version of the Jupyter app for course environments
* [Jupyter](jupyter.md) Interactive computational environment with Python
* [Kaldi](http://urn.fi/urn:nbn:fi:lb-2019121020) Kaldi Speech Recognition Toolkit
* [Kraken2](kraken.md) Taxonomic sequence classification system
* [Krona](krona.md) Visualization tool for taxonomic classification and other hierarchical data
* [LAMMPS](lammps.md) Fast molecular dynamics engine with large force field selection
* [LAStools](lastools.md) for LiDAR datasets
* [Lazypipe](lazypipe.md) A stand-alone pipeline for identifying viruses in host-associated or environmental samples
* [MACS2](macs2.md) ChIP-Seq analysis tool
* [Maestro](maestro.md) Versatile drug discovery and materials modeling suite
* [Materials Studio](materialsstudio.md) Versatile materials modeling suite
* [MATLAB](matlab.md) High-level technical computing language
* [MaxQuant](../support/tutorials/MaxQuant-tutorial.md) A proteomics software for processing of Mass-spectromtery data
* [Megahit](megahit.md) Metagenomics assembly
* [MetaPhlAn](metaphlan.md) Profiling the composition of microbial communities with metagenomic data
* [Minimap2](minimap2.md) Short read aligner
* [MIRA](mira.md) Whole genome shotgun and EST sequence Assembler
* [MOLDEN](molden.md) Processing program for molecular and electronic structure calculations
* [MOLPRO](molpro.md) Package for accurate ab initio quantum chemistry calculations
* [Mothur](mothur.md) Package for microbial community analysis of amplicon sequencing data
* [MrBayes](mrbayes.md) Program for inferring phylogenies using Bayesian methods
* [MySQL](../data/kaivos/overview.md) MySQL database client.
* [NAMD](namd.md) Highly scalable classical molecular dynamics
* [Nextflow](nextflow.md) Nextflow is a scientific workflow management system for creating scalable, portable, and reproducible workflows
* [NWChem](nwchem.md) A computational chemistry software package designed to perform well on parallel HPC systems
* [Octave](octave.md) High-level interpreted language for numerical computations
* [Open Babel](openbabel.md) Program to interconvert file formats currently used in molecular modeling
* [OpenDroneMap](opendronemap.md) for processing aerial drone imagery
* [OpenFOAM](openfoam.md) Open source C++ tool box for continuum mechanics problems
* [ORCA](orca.md) General purpose quantum chemistry package
* [Orfeo ToolBox](otb.md) for remote sensing applications
* [PALM](palm.md) Meteorological model system for atmospheric and oceanic boundary-layer flows
* [Pannzer](pannzer.md) Automatic protein annotation tool
* [ParaView](paraview.md) Free open-source visualization application
* [PCL](pcl.md) for 2D/3D image and point cloud processing
* [PDAL](pdal.md) for point cloud translations and processing
* [Pennylane](pennylane.md) Free open-source software framework for quantum machine learning and quantum computing
* [Picard](picard.md) Tools for working with SAM,BAM,CRAM and VCF files
* [Plumed](plumed.md) Library and tools for enhanced sampling methods
* [Prokka](prokka.md) Rapid prokaryotic genome annotation
* [Python Data](python-data.md) Collection of Python libraries for data analytics and machine learning
* [Python](python.md) programming language and its modules at CSC
* [PyTorch](pytorch.md) Machine learning framework for Python
* [QGIS](qgis.md) General purpose GIS software family for viewing, editing and analysing geospatial data
* [QIIME2](qiime.md) Package for microbial community analysis of amplicon sequencing data
* [Qiskit](qiskit.md) open-source toolkit for useful quantum computing
* [RAPIDS](rapids.md) Suite of libraries for data analytics and machine learning on GPUs
* [RAxML](raxml.md) Program for inferring phylogenies with likelihood
* [r-env](r-env.md) R, RStudio Server, SAGA and TensorFlow
* [R for GIS](r-env-for-gis.md) R spataial analysis libraries
* [Roary](roary.md) Pan genome pipeline
* [RStudio](rstudio.md) Integrated development environment for R
* [SAGA GIS](saga-gis.md) General purpose GIS software family for viewing, editing and analysing geospatial data
* [SageMath](sagemath.md) Free open-source mathematics software system
* [Salmon](salmon.md) Program to produce transcript-level quantification estimates from RNA-seq data
* [Samtools](samtools.md) Utilities for managing SAM/BAM formatted alignment files
* [Seismic-unix](seismic-unix.md) for processing of 2D seismic or GPR data sets.
* [Sen2Cor](sen2cor.md) for atmospheric-, terrain and cirrus correction of the Sentinel-2 products
* [Sen2mosaic](sen2mosaic.md) for download, preprocessing and mosaicing of Sentinel-2 products
* [Seqtk](seqtk.md) Tool for processing sequences in the FASTA or FASTQ format
* [SNAP](snap.md) for remote sensing applications
* [SOFI3D](sofi3d.md) for 3D finite-difference seismic wave simulation
* [SPAdes](spades.md) Genome assembly
* [Spark](spark.md) High-performance distributed computing framework
* [Stacks](stacks.md) Pipeline for building loci from short-read sequences (e.g. RAD-seq data)
* [Star-CCM+](starccm+.md) Computational Fluid Dynamics software by Siemens Digital Industries Software
* [STAR](star.md) Short read aligner
* [Structure](structure.md) Inference of population structure in genetics
* [TensorBoard](tensorboard.md) The visualization toolkit for TensorFlow
* [TensorFlow](tensorflow.md) Deep learning library for Python
* [TmoleX](tmolex.md) GUI for setting up and analyzing TURBOMOLE jobs
* [Tophat](tophat.md) Splice junction mapper for RNA-Seq reads
* [Trinity](trinity.md) Traskriptome assembly tool
* [Turbomole](turbomole.md) Efficient program package for electronic structure calculations
* [UDPipe](http://urn.fi/urn:nbn:fi:lb-201902131) UDPipe Kielipankki version
* [VASP](vasp.md) Ab initio DFT electronic structures
* [Velvet](velvet.md) Genome assembler
* [VirusDetect](virusdetect.md) Virus identification with sRNA data
* [VisIt](visit.md) Free open-source visualization application
* [Visual Studio Code](vscode.md) Source code editor
* [VMD](vmd.md) Molecular visualization program
* [WhiteboxTools](whiteboxtools.md) an advanced geospatial data analysis platform
* [wtdbg2](wtdbg2.md) Fast assembler for long-read data
* [XHMM](xhmm.md) Copy number variation calling from targeted sequencing data
* [Zonation](zonation.md) Spatial conservation prioritization framework
